codexctl/cpio.py

```
@@ -0,0 +1,150 @@
+def _parse_cpio_header(header_data: bytes) -> Dict[str, int]:
+    """
+    Parse a CPIO newc format header (110 bytes).
+
+    Format: magic(6) + inode(8) + mode(8) + uid(8) + gid(8) + nlink(8) +
+            mtime(8) + filesize(8) + devmajor(8) + devminor(8) +
+            rdevmajor(8) + rdevminor(8) + namesize(8) + check(8)
+
+    All fields except magic are 8-digit ASCII hex.
+    """
+    if len(header_data) != 110:
+        raise ValueError(f"Invalid header size: {len(header_data)} (expected 110)")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)